x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/NodeSeenService.java

```diff
@@ -46,8 +46,11 @@ public void clusterChanged(ClusterChangedEvent event) {
             return;
         }
 
-        if (event.nodesAdded() == false) {
-            // If there's no new nodes this cluster state update, nothing to do.
+        final boolean thisNodeJustBecameMaster = event.previousState().nodes().isLocalNodeElectedMaster() == false
+            && event.state().nodes().isLocalNodeElectedMaster();
+        if ((event.nodesAdded() || thisNodeJustBecameMaster) == false) {
+            logger.error("GWB> Bailing early");
```

Comment:
We should probably remove this line :)

Reply:
Whoops, I had those in there to help verify my reading that the `== false` was missing 🤦 

Removed.